src/gluonts/zebras/_time_frame.py

```diff
@@ -136,6 +136,76 @@ def pad(self, value, left: int = 0, right: int = 0) -> TimeFrame:
             _pad=Pad(pad_left, pad_right),
         )
 
+    def update(self, other: TimeFrame, default=np.nan) -> TimeFrame:
+        """Create a new ``TimeFrame`` which includes values of both input
+        frames.
+
+        The new frame spans both input frames and inserts default values if
+        there is a gap between the two frames. If both frames overlap, the
+        second can overwrite the values of the first frame.
+
+        Static columns and metadata is also updated, and the second frames
+        value take precedence.
+
+        Updating a frame with itself is effectively a noop, similar to how
+        ``dict.update`` on the same dict will return an identical result.
+
+        Update requires that both frame have defined indices, since otherwise
+        its not possible to know how the values relate to each other.
+
+        Note: ``update`` will reset the padding.
+        """
+
+        assert self.index is not None and other.index is not None
+        assert self.index.freq == other.index.freq
+
+        start = min(self.index.start, other.index.start)
+        end = max(self.index.end, other.index.end)
+
+        index = Periods(
+            np.arange(
+                start.to_numpy(),
+                # arange is exclusive, thus we need to add `1`
+                (end + 1).to_numpy(),
+                start.freq.step,
+            ),
+            start.freq,
+        )
+        self_idx0 = index.index_of(self.index.start)
+        other_idx0 = index.index_of(other.index.start)
+
+        columns = {}
+        for name, (self_col, other_col, tdim) in rows_to_columns(
+            (self.columns, other.columns, self.tdims)
+        ).items():
```

Comment:
What columns are taken? Is it assumed that the two frames have the same columns? On only the ones from `self` are considered?

Reply:
Yes, the idea is that they need to be identical at the moment. I guess what we can do is to allow to update the result only partially and just fall back on the default value. That way one can even update different parts of the time frame at different times.